src/components/textInput/TextInputBase.tsx

```diff
@@ -0,0 +1,348 @@
+    fontWeight: "400",
+    lineHeight: 24,
+    fontFamily: "Readex Pro",
+    height: "100%"
+  },
+  textInputLabelWrapper: {
+    position: "absolute",
+    paddingHorizontal: 10,
+    zIndex: 10,
+    bottom: 0,
+    top: 0,
+    justifyContent: "center"
+  }
+});
+
+type InputTextHelperRow = {
+  value: string;
+  counterLimit?: number;
+  helperText?: string;
+  errorMessage?: string;
+  isValid?: boolean;
+};
+
+const HelperRow = ({
+  value,
+  counterLimit,
+  helperText,
+  errorMessage,
+  isValid
+}: InputTextHelperRow) => {
+  const valueCount = useMemo(() => value.length, [value]);
+
+  const helperRowStyle: ViewStyle = useMemo(() => {
+    if (counterLimit && helperText) {
+      return {
+        justifyContent: "space-between"
+      };
+    }
+    if (counterLimit) {
+      return {
+        justifyContent: "flex-end"
+      };
+    }
+    if (helperText) {
+      return {
+        justifyContent: "flex-start"
+      };
+    }
+    return {};
+  }, [counterLimit, helperText]);
+
+  return (
+    <View
+      style={[
+        IOStyles.row,
+        {
+          alignItems: "center",
+          paddingHorizontal: 10
+        },
+        helperRowStyle
+      ]}
+    >
+      {errorMessage && isValid === false && (
+        <LabelSmall weight="Regular" color="error-600">
+          {errorMessage}
+        </LabelSmall>
+      )}
+      {helperText && !(errorMessage && isValid === false) && (
+        <LabelSmall weight="Regular" color="grey-700">
+          {helperText}
+        </LabelSmall>
+      )}
+      {counterLimit && (
+        <LabelSmall
+          weight="Regular"
+          color="grey-700"
+        >{`${valueCount} / ${counterLimit}`}</LabelSmall>
+      )}
+    </View>
+  );
+};
+
+export const TextInputBase = ({
+  disabled = false,
+  placeholder,
+  value = "",
+  icon,
+  counterLimit,
+  helperText,
+  onChangeText,
+  onValidate,
+  secretInput,
+  errorMessage
+}: InputTextProps) => {
+  const labelSharedValue = useSharedValue(0);
+  const [inputStatus, setInputStatus] = React.useState<InputStatus>(
+    disabled ? "disabled" : "initial"
+  );
+  const [isValid, setIsValid] = React.useState<boolean | undefined>(undefined);
+  const [isSecretInput, setSecretInput] = React.useState<boolean>(
+    secretInput ?? false
+  );
+
+  const inputRef = useRef<TextInput>(null);
+
+  const boxStyle: ViewStyle = useMemo(() => {
+    if (inputStatus === "focused") {
+      return {
+        borderColor: IOColors["blueIO-500"],
+        borderWidth: 2
+      };
+    }
+    if (inputStatus === "error") {
+      return {
+        borderColor: IOColors["error-600"],
+        borderWidth: 1
+      };
+    }
+    return {
+      borderColor: IOColors["grey-200"],
+      borderWidth: 1
+    };
+  }, [inputStatus]);
+
+  const animatedLabelProps = useAnimatedStyle(() => ({
+    fontSize: withTiming(interpolate(labelSharedValue.value, [0, 1], [16, 12])),
+    top: withTiming(interpolate(labelSharedValue.value, [0, 1], [0, -16]))
```

Comment:
I don't really like magic numbers without explanation, but:
- `16` is too close to the edge
- `12` is good, but it starts to overlap the input caret
```suggestion
    transform: [
      {
        translateY: withTiming(
          interpolate(labelSharedValue.value, [0, 1], [0, -14]),
          {
            duration: 300,
            easing: Easing.elastic(0.85)
          }
        )
      }
    ]
```
Also, I'm probably biased by the best practices about Web animation performance (maybe not valid in the RN layout engine), but applying the transition using `translateY` should be more performant and also clearer to the reader